### ISSUE TITLE
Ingress Gateway Pod Not Ready

### ISSUE THREAD

**Issue by liuxu623** [istio/istio#32499]
istio version 1.6.9

I found some ingress gateway pods not ready

```
kubectl get pod -n istio-system
NAME                                       READY   STATUS    RESTARTS   AGE
debug                                      1/1     Running   0          169d
istio-ingressgateway-7c4977d74-29q8h       1/1     Running   0          3d7h
istio-ingressgateway-7c4977d74-2br9d       1/1     Running   0          4d19h
istio-ingressgateway-7c4977d74-bw5lz       0/1     Running   0          5d7h
istio-ingressgateway-7c4977d74-kl2k5       1/1     Running   0          2d4h
istio-ingressgateway-7c4977d74-v4tq9       1/1     Running   0          29h
```
and logs 
```
2021-04-28T04:53:16.312780Z	warn	Envoy proxy is NOT ready: failed to get readiness stats: Get "http://127.0.0.1:15000/stats?usedonly&filter=^(server\\.state|listener_manager\\.workers_started)": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
```

envoy admin thread cpu ~100%

![image](https://user-images.githubusercontent.com/9653438/116348742-387c7600-a821-11eb-8ab5-20502dc11bb7.png)

15000 port not work include `curl http://localhost:15000/ready` and `curl http://localhost:15000/server_info`

![image](https://user-images.githubusercontent.com/9653438/116348795-5053fa00-a821-11eb-85ba-204024abb731.png)

I think it maybe related to we use Envoyfilter  add url_path label to prometheus metrics

```yaml
  - applyTo: HTTP_FILTER
    match:
      context: GATEWAY
      listener:
        filterChain:
          filter:
            name: envoy.http_connection_manager
            subFilter:
              name: envoy.router
      proxy:
        proxyVersion: ^1\.6.*
    patch:
      operation: INSERT_BEFORE
      value:
        name: istio.stats
        typed_config:
          '@type': type.googleapis.com/udpa.type.v1.TypedStruct
          type_url: type.googleapis.com/envoy.extensions.filters.http.wasm.v3.Wasm
          value:
            config:
              configuration: |
                {
                  "debug": "false",
                  "stat_prefix": "istio",
                  "disable_host_header_fallback": true,
                  "metrics": [
                      {
                        "name": "requests_total",
                        "dimensions": {
                          "destination_port": "string(destination.port)",
                          "request_host": "request.host",
                          "request_url_path": "request.url_path"
                        }
                      },
                      {
                        "name": "request_duration_milliseconds",
                        "dimensions": {
                          "destination_port": "string(destination.port)",
                          "request_host": "request.host",
                          "request_url_path": "request.url_path"
                        }
                      },
                      {
                        "name": "request_bytes",
                        "dimensions": {
                          "destination_port": "string(destination.port)",
                          "request_host": "request.host",
                          "request_url_path": "request.url_path"
                        }
                      },
                      {
                        "name": "response_bytes",
                        "dimensions": {
                          "destination_port": "string(destination.port)",
                          "request_host": "request.host",
                          "request_url_path": "request.url_path"
                        }
                      }
                    ]
                }
              root_id: stats_outbound
              vm_config:
                code:
                  local:
                    inline_string: envoy.wasm.stats
                runtime: envoy.wasm.runtime.null
                vm_id: stats_outbound
```

**Comment by liuxu623** [comment]
@ramaraochavali @ramaraochavali
Could we use `http://localhost:15000/ready` instead of `http://127.0.0.1:15000/stats?usedonly&filter=^(server\\.state|listener_manager\\.workers_started)`?
Maybe too heavy use stats to check the health status of envoy.
And if we have too much metrics, stats will be slow.

**Comment by howardjohn** [comment]
https://istio.io/latest/news/support/announcing-1.6-eol/

Please upgrade, this is already fixed in 1.9. We now cache the initial response so the stats call is trivial. 

Thanks!

**Comment by liuxu623** [comment]
> https://istio.io/latest/news/support/announcing-1.6-eol/
> 
> Please upgrade, this is already fixed in 1.9. We now cache the initial response so the stats call is trivial.
> 
> Thanks!

I found https://github.com/istio/istio/pull/26820 thank you.
